### PR TITLE
Rchain 3746 slash justification bonded

### PR DESCRIPTION
**Pull request by zsluedem**
## Overview
<sup>_What this PR does, and why it's needed_</sup>

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3746

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by zsluedem** [comment]
bors try

File: integration-tests/test/test_slash.py [hunk 1]
```diff
@@ -120,3 +120,47 @@ def test_slash_invalid_block_seq(command_line_options: CommandLineOptions, rando
...
+        invalid_justifications_block.CopyFrom(block_msg)
+        error_justification = Justification(validator=PrivateKey.generate().to_bytes(), latestBlockHash=block_msg.blockHash)
+
+        invalid_justifications_block.justifications.append(error_justification)  # pylint: disable=maybe-no-member
+        # change timestamp to make block hash different
+        invalid_justifications_block.header.timestamp = block_msg.header.timestamp + 1  # pylint: disable=maybe-no-member
+        invalid_block_hash = gen_block_hash_from_block(invalid_justifications_block)
+        invalid_justifications_block.sig = BONDED_VALIDATOR_KEY_1.sign_block_hash(invalid_block_hash)
+        invalid_justifications_block.blockHash = invalid_block_hash
+        import pdb;pdb.set_trace()
```
Comment (KentShikama) [inline integration-tests/test/test_slash.py]: You'll need to remove this line

File: integration-tests/test/test_slash.py [hunk 2]
```diff
@@ -120,3 +120,47 @@ def test_slash_invalid_block_seq(command_line_options: CommandLineOptions, rando
         bonds_validators = extract_validator_stake_from_bonds_validator_str(block_info['bondsValidatorList'])
 
         assert bonds_validators[BONDED_VALIDATOR_KEY_1.get_public_key().to_hex()] == 0.0
+
+@pytest.mark.skipif(sys.platform in ('win32', 'cygwin', 'darwin'), reason="Only Linux docker support connection between host and container which node client needs")
+def test_slash_justification_not_correct(command_line_options: CommandLineOptions, random_generator: Random, docker_client: DockerClient) -> None:
+    """
+    Slash a validator which proposed a block with justifications not matching bonded validators
```
Comment (KentShikama) [inline integration-tests/test/test_slash.py]: Slash a validator which proposed a block with justifications not matching bonded validators of main parent

File: integration-tests/test/test_slash.py [hunk 3]
```diff
@@ -120,3 +120,47 @@ def test_slash_invalid_block_seq(command_line_options: CommandLineOptions, rando
...
+    """
+    bonded_validators = {
+        BOOTSTRAP_NODE_KEYS: 100,
```
Comment (KentShikama) [inline integration-tests/test/test_slash.py]: KEY?

**Comment by zsluedem** [comment]
bors try

**Review by woky — APPROVED** [review]
Python looks OK

**Comment by zsluedem** [comment]
bors try

**Review by adaszko — APPROVED** [review]

**Pull request merged**